File: OOPs/_4_Polymorphism/polymorphism.dart
class Animal {
  void makeSound() {
    print('Some animal makes a sound');
  }
}

class Dog extends Animal {
  @override
  void makeSound() {
    print('Dog barks');
  }
}

class Cat extends Animal {
  @override
  void makeSound() {
    print('Cat meows');
  }
}

void main() {
  List<Animal> animals = [Dog(), Cat()];
  
  for (var animal in animals) {
    animal.makeSound();  // Dog barks, Cat meows
  }
}

File: OOPs/_4_Polymorphism/polymorphism.py
class Animal:
    def make_sound(self):
        print('Some animal makes a sound')

class Dog(Animal):
    def make_sound(self):
        print('Dog barks')

class Cat(Animal):
    def make_sound(self):
        print('Cat meows')

animals = [Dog(), Cat()]

for animal in animals:
    animal.make_sound()  # Dog barks, Cat meows
